### DISCUSSION TITLE
Inputs

### DISCUSSION THREAD

**Discussion by crazed-developer** [quarkiverse/quarkus-jef#80] [General]
I have a question about inputs. 

It has been straightforward to turn a LED on and off via the modified example app (This is outputs):

https://github.com/alexeysharandin/quarkus-jef-example

Running on a Raspberry Pi 4, I had to disable/delete everything except for the GPIO..

Reacting to inputs have been a struggle though, I can't figure it out.. Been reading through the source code and I see no solutions.

Is it possible, if yes. Can you supply an example ?

Thank you for the good work!

**Comment by alexeysharandin** [comment]
Hi Steffen,
not sure what I understand a problem.
Could you please provide a some reproducer or more detalis.
Looks like you using GPIOs what mapped to some other interfaces like I2C/SPI/SDIO and GPIO in parallel or maybe have invallid connection for LED.

**Reply by crazed-developer** [reply]
Hi Alexey,

Sorry if my question was a little muddy. 

LED control are working fine on the GPIO using your code.

My question are about handling inputs, like creating code to react to a button press. I see no such example code, and was unable to figure out how (if possible) looking at the JEF source code itself.

**Reply by alexeysharandin** [reply]
you need to set pin.setDirection to INPUT and read value in some scheduled method over https://quarkus.io/guides/scheduler-reference for example

**Reply by crazed-developer** [reply]
I feel that your suggestion could work for more like a switch scenario, where it's either turned on or turned off, for longer periods of time. 

But for a button, where you push it on, then release and it goes off again. I guess that there would be a big chance of the scheduler doing the check in-between the push. And thus miss the push. 


Having a look at this project: https://pi4j.com/examples/components/simplebutton/

They provide an eventhandler for: onDown, onUp and whilePressed. 

Being able to attach an eventhandler would be very useful. 


In any case, thanks for your help, and your work on the project.

**Reply by alexeysharandin** [reply]
The are no API in linux to support events. In pi4j - author duing the same what I propose above - https://github.com/Pi4J/pi4j-example-components/blob/main/src/main/java/com/pi4j/catalog/components/SimpleButton.java#L52C9-L52C9
but creating separated thread per each IN gpio pin what not a good.

**Reply by crazed-developer** [reply]
Alexey,

Thank you for your dialogue, it's been positive for me. I will look further into this method.

Have a nice day!

Steffen
